--- a/nf_nat_fullcone.c
+++ b/nf_nat_fullcone.c
@@ -1130,7 +1130,7 @@ static unsigned int nf_nat_handle_prerouting(u8 nfproto, struct sk_buff *skb, un
 	enum ip_conntrack_info ctinfo;
 	struct nf_conntrack_tuple *ct_tuple_origin;
 
-	uint16_t port, original_port;
+	uint16_t port;
 	uint8_t protonum;
 
 /* NFPROTO specific def */
@@ -1151,7 +1151,6 @@ static unsigned int nf_nat_handle_prerouting(u8 nfproto, struct sk_buff *skb, un
 	ip_6 = NULL;
 	/* NFPROTO specific init end */
 
-	original_port = 0;
 	ret = NFT_CONTINUE;	// BUG: use XT_CONTINUE for Xtables
 
 	ct = nf_ct_get(skb, &ctinfo);
@@ -1265,9 +1264,9 @@ static unsigned int nf_nat_handle_postrouting(u8 nfproto, struct sk_buff *skb, u
 	enum ip_conntrack_info ctinfo;
 	struct nf_conn_nat *nat;
 	struct nf_conntrack_tuple *ct_tuple, *ct_tuple_origin;
-	uint16_t port, original_port, want_port;
+	uint16_t port, original_port, want_port = 0;
 	uint8_t protonum;
-	bool is_src_mapping_active;
+	bool is_src_mapping_active = false;
 
 	/* NFPROTO specific def */
 	struct nat_mapping *mapping, *src_mapping;
